libdnf/sack/query.hpp

```diff
@@ -179,6 +186,16 @@ struct Query {
     std::unique_ptr<Impl> pImpl;
 };
 
+inline Query::ExcludeFlags operator|(Query::ExcludeFlags a, Query::ExcludeFlags b)
+{
+    return static_cast<Query::ExcludeFlags>(static_cast<int>(a) | static_cast<int>(b));
+}
+
+inline bool operator&(Query::ExcludeFlags a, Query::ExcludeFlags b)
+{
+    return (static_cast<int>(a) & static_cast<int>(b)) == static_cast<int>(b);
+}
```

Comment:
The signature and semantics of the bit logic operators should surely be the same (even though for specifically these flag enums `|` is typically used for adding flags together and `&` is used for checking of presence, I think this is really unexpected and confusing).

Returning `Query::ExcludeFlags` seems cleanest, but then type safety really gets in the way of practical use of `&` for checking the flags. Short of defining a function or overloading a different, special operator that would return `bool` I can't think of a way around it.